conn/controlfns_unix.go

```diff
@@ -0,0 +1,23 @@
+//go:build !windows && !linux && !js
+
+/* SPDX-License-Identifier: MIT
+ *
+ * Copyright (C) 2017-2022 WireGuard LLC. All Rights Reserved.
+ */
+
+package conn
+
+import "syscall"
+
+func init() {
+	controlFns = append(controlFns,
+		// Set SO_RCVBUF/SO_SNDBUF - this could be common with the _unix code except
+		// for the unfortunate type specificity of syscall.Handle.
+		func(network, address string, c syscall.RawConn) error {
+			return c.Control(func(fd uintptr) {
```

Comment:
This doesn't need IPV6_V6ONLY on the v6 socket for BSDs?

Reply:
Sticky sockets was only ever implemented for Linux, and that remains true after this patch. I think this patch carries most of the necessary code to turn on sticky socket support on Darwin and the BSD's, and with a little bit more on Windows too - but we will need to do independent testing of that feature set across the platforms, so I'd prefer to do it separately.

Reply:
`IPV6_V6ONLY` is independent of sticky sockets, though. It's so that you get v4 connections on the v4 socket and don't wind up with endpoints set to v6-mapped-v4 addresses.

Reply:
Added `IPV6_V6ONLY` socket option for BSDs in https://github.com/WireGuard/wireguard-go/pull/64/commits/f5ff18eb271ae0cab157aec570bc1096ca7fae8d.